packages/ui/src/components/SelectableCardGroup/index.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from '@emotion/styled'
+import { Icon } from '@ultraviolet/icons'
+import {
+  type ComponentProps,
+  type InputHTMLAttributes,
+  type ReactNode,
+  createContext,
+  useContext,
+  useMemo,
+} from 'react'
+import { Row } from '../Row'
+import { SelectableCard } from '../SelectableCard'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+
+type SelectableCardGroupContextType = {
+  groupName: string
+  groupValue: string | number | (string | number)[]
+  type: 'radio' | 'checkbox'
+  showTick: boolean
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange'>> &
+  Pick<InputHTMLAttributes<HTMLInputElement>, 'required'>
+
+const SelectableCardGroupContext = createContext<
+  SelectableCardGroupContextType | undefined
+>(undefined)
+
+type CardSelectableCardProps = Omit<
+  ComponentProps<typeof SelectableCard>,
+  'onChange' | 'checked' | 'type' | 'showTick'
+>
+
+const CardSelectableCard = ({
+  value,
+  disabled,
+  children,
+  className,
+  isError,
+  onFocus,
+  onBlur,
+  tooltip,
+  id,
+  label,
+  'data-testid': dataTestId,
+}: CardSelectableCardProps) => {
+  const context = useContext(SelectableCardGroupContext)
+
+  if (!context) {
+    throw new Error(
+      'SelectableCardGroup.Card can only be used inside a SelectableCardGroup',
+    )
+  }
+
+  const { groupName, onChange, groupValue, type, showTick } = context
+
+  return (
+    <SelectableCard
+      name={groupName}
+      value={value}
+      showTick={showTick}
+      type={type}
+      disabled={disabled}
+      className={className}
+      isError={isError}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      tooltip={tooltip}
+      id={id}
+      label={label}
+      onChange={onChange}
+      checked={
+        typeof groupValue === 'object'
+          ? groupValue.includes(value)
+          : groupValue === value
+      }
+      data-testid={dataTestId}
+    >
+      {children}
+    </SelectableCard>
+  )
+}
+
+const FieldSet = styled.fieldset`
+  border: none;
+  padding: 0;
+  margin: 0;
+`
+
+const StyledRequiredIcon = styled(Icon)`
+  vertical-align: super;
+`
+
+type SelectableCardGroupProps = {
+  legend: string
```

Comment:
We should make legend optional here cause we have plenty of use case without any legend, I know it is not how we did for radio and checkbox 😅